api.js

```diff
@@ -37,6 +38,27 @@ class Api extends Emittery {
 
 		this.options = Object.assign({match: []}, options);
 		this.options.require = resolveModules(this.options.require);
+
+		const {extensions: doNotCompileExtensions = []} = this.options;
+		const {extensions: babelExtensions = []} = this.options.babelConfig || {};
+
+		if (!this.options.babelConfig && doNotCompileExtensions.length === 0) {
+			doNotCompileExtensions.push('js');
+		}
+
+		if (this.options.babelConfig && babelExtensions.length === 0) {
+			babelExtensions.push('js');
+		}
+
+		this.options.extensions = doNotCompileExtensions;
+
+		// Combine all extensions possible for testing. Removing duplicate extensions.
+		this.allExtensions = [...new Set([...doNotCompileExtensions, ...babelExtensions])];
+
+		if (this.allExtensions.length !== doNotCompileExtensions.length + babelExtensions.length) {
+			const duplicates = difference([...doNotCompileExtensions, ...babelExtensions]);
```

Comment:
You need to compare two arrays for `difference` to return anything. But then it actually gives the extensions that are not repeated, so:

```js
const notRepeated = difference(doNotCompileExtensions, babelExtensions);
const duplicates = this.allExtensions.filter(ext => !notRepeated.includes(ext));
```

I suppose this still needs a test?